src/main/groovy/edu/oregonstate/mist/slashcourses/resources/SlashCourseResource.groovy

```diff
@@ -74,16 +77,75 @@ class SlashCourseResource extends Resource {
-//        ResponseBuilder responseBuilder = ok(message)
-//        responseBuilder.build()
-//    }
+    @Path('{crn: \\d+}')
+    @DELETE
+    @Produces(MediaType.APPLICATION_JSON)
+    public Response deleteByCRN (@PathParam('crn') Integer crn) {
+        slashCourseDAO.deleteByCRN(crn)
+        Response.ok().build()
+    }
+
+    @POST
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Produces(MediaType.APPLICATION_JSON)
+    public Response postCourse (@Valid SlashCourse newCourse) {
+
+        Response returnResponse
+        try {
+
+            List<Integer> crnList = slashCourseDAO.getAllCRN()
+
+            if (newCourse.getInstructor() != null && !crnList.contains(newCourse.getCrn())) {
+                instructorDAO.postInstructor(newCourse.getInstructor().getLastName(),
+                                             newCourse.getInstructor().getFirstName())
+                slashCourseDAO.postCourse(newCourse.getCrn(),
+                                          newCourse.getCourseNum(),
+                                          newCourse.getCourseName(),
+                                          newCourse.getSlash(),
+                                          newCourse.getTerm(),
+                                          // retrieve the instructorId just created as instructorId in the new course object
+                                          instructorDAO.getLatestInstructorId().toInteger(),
+                                          newCourse.getDay(),
+                                          newCourse.getTime(),
+                                          newCourse.getLocation(),
+                                          newCourse.getType(),
+                                          newCourse.getInstructor())
+
+                URI createdURI = URI.create("/" + instructorDAO.getLatestInstructorId())
+                returnResponse = Response.created(createdURI).build()
+            } else {
+                slashCourseDAO.postCourse(newCourse.getCrn(),
+                                          newCourse.getCourseNum(),
+                                          newCourse.getCourseName(),
+                                          newCourse.getSlash(),
+                                          newCourse.getTerm(),
+                                          newCourse.getInstructorId(),
+                                          newCourse.getDay(),
+                                          newCourse.getTime(),
+                                          newCourse.getLocation(),
+                                          newCourse.getType(),
+                                          newCourse.getInstructor())
+                returnResponse = Response.ok().build()
+            }
+
+        } catch (Exception e) {
+            String constraintError = e.cause.toString()
+            System.out.println("*** DEBUG " + constraintError + "***")
```

Comment:
I believe dropwizard provides you with a log method that you can use to print out debug statements :) . These debug statements can then be toggled on / off via the log evel in the configuration file.